controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -458,6 +459,16 @@ func (r *Reconciler) checkIsTargetValid(ctx *pkgctx.VirtualMachinePublishRequest
 			return err
 		}
 		contentLibraryType = contentLibraryV1A2.Spec.Type
+
+		if contentLibraryType == imgregv1.LibraryTypeInventory && contentLibraryV1A2.Labels != nil {
```

Comment:
I think metav1.HasLabel() checks for nil as well.

Reply:
it doesn't